node/src/components/rpc_server.rs

```diff
@@ -294,13 +294,16 @@ where
                 }
                 .ignore()
             }
-            Event::RpcRequest(RpcRequest::GetLowestContiguousBlockHeight { responder }) => {
+            Event::RpcRequest(RpcRequest::GetHighestContiguousBlockHeightRange { responder }) => {
                 effect_builder
-                    .get_lowest_contiguous_block_height_from_storage()
-                    .event(move |height| Event::GetLowestContiguousBlockHeightResult {
-                        height,
-                        main_responder: responder,
-                    })
+                    .get_highest_contiguous_block_range_from_storage()
+                    .event(
+                        move |(low, high)| Event::GetHighestContiguousBlockHeightRange {
+                            low,
+                            high,
+                            main_responder: responder,
+                        },
+                    )
```

Comment:
Couldn't you simply map over the result and call `responder.respond((low, high))` ? It would require adding `and_then` on the `EffectBuilder` if I understand correctly.

Reply:
Yes, it could actually work as follows: https://github.com/casper-network/casper-node/pull/2786/commits/bcbbb7b05a8cd3c581b65908ce22c4f0cc3ce959

Is it better now, @goral09 ?

Reply:
Not the prettiest thing but achieves what I meant :)